### PR TITLE
Fix Mistake in convert_fO2_to_fe_partition

### PR DESCRIPTION
**Pull request by fboschetty**
Fixed reference to 'logfo2_offset', which should read 'fo2_offset' around line 4172.
Updated formatting to improve readability

### CODE REVIEW

**Comment by PennyWieser** [comment]
Hi Felix, I independently was working on a bug fix for this. I cant follow exactly what you changed as a lot of the changes are just entering of new lines which is making it super hard to follow. I'm hoping my fix is the same as your fix? But let me know if not. I also added some more buffers.

**Comment by fboschetty** [comment]
Hi Penny,

The issue was on line 4172. The code references 'logfo2_offset' which doesn't exist. It should be 'fo2_offset' instead. Doesn't look like that's changed as of today.

Felix
